pandas/tests/io/pytables/test_path.py

```diff
@@ -0,0 +1,163 @@
+import os
+import pytest
+
+import pandas as pd
+import pandas.util.testing as tm
+from pandas.tests.io.pytables.common import ensure_clean_path
+from pandas.compat import PY36
+from pandas.io import pytables as pytables
+from pandas.io.pytables import HDFStore, ClosedFileError
+
+
+def test_path_pathlib():
+    df = tm.makeDataFrame()
+
+    result = tm.round_trip_pathlib(
+        lambda p: df.to_hdf(p, 'df'),
+        lambda p: pd.read_hdf(p, 'df'))
+    tm.assert_frame_equal(df, result)
+
+
+def test_path_pathlib_hdfstore():
+    df = tm.makeDataFrame()
+
+    def writer(path):
+        with pd.HDFStore(path) as store:
+            df.to_hdf(store, 'df')
+
+    def reader(path):
+        with pd.HDFStore(path) as store:
+            return pd.read_hdf(store, 'df')
+
+    result = tm.round_trip_pathlib(writer, reader)
+    tm.assert_frame_equal(df, result)
+
+
+def test_pickle_path_localpath():
+    df = tm.makeDataFrame()
+    result = tm.round_trip_pathlib(
+        lambda p: df.to_hdf(p, 'df'),
+        lambda p: pd.read_hdf(p, 'df'))
+    tm.assert_frame_equal(df, result)
+
+
+def test_path_localpath_hdfstore():
+    df = tm.makeDataFrame()
+
+    def writer(path):
+        with pd.HDFStore(path) as store:
+            df.to_hdf(store, 'df')
+
+    def reader(path):
+        with pd.HDFStore(path) as store:
+            return pd.read_hdf(store, 'df')
+
+    result = tm.round_trip_localpath(writer, reader)
+    tm.assert_frame_equal(df, result)
+
+
+def test_multiple_open_close():
+    # gh-4409: open & close multiple times
+    with ensure_clean_path() as path:
+
+        df = tm.makeDataFrame()
+        df.to_hdf(path, 'df', mode='w', format='table')
+
+        # single
+        store = HDFStore(path)
+        assert 'CLOSED' not in store.info()
+        assert store.is_open
+
+        store.close()
+        assert 'CLOSED' in store.info()
+        assert not store.is_open
```

Comment:
these need to go with any other open/close tests, which I think you have in test_common.py now. call this file test_open_close (move this one and other tests)